llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -112,6 +112,8 @@ def __init__(
                 else:
                     setattr(config, k, v)
 
+            load_in_8bit = om_model_config.get('load_in_8bit', False)
```

Comment:
Since this doesn't work with FSDP, can we check somewhere and raise a nice error if the user tries to combine 8bit + fsdp? Also, since you've only tried this for eval (correct me if i'm wrong), we should probably raise a nice error if a user tries to train with this on. Maybe we just do some checking in `train.py` and `eval.py`? What do you think?

Reply:
Makes sense, I've added checks for fsdp in `eval.py` and `load_in_8bit` in `train.py` and tested the exceptions now.